src/SubPoolFactory.sol

```diff
@@ -0,0 +1,193 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.26;
+
+import {Auth} from "./Auth.sol";
+import {SafeTransferLib} from "solady/utils/SafeTransferLib.sol";
+import {SubPool} from "./SubPool.sol";
+import {ISubPoolFactory} from "./interfaces/ISubPoolFactory.sol";
+
+using SafeTransferLib for address;
+
+contract SubPoolFactory is Auth, ISubPoolFactory {
+    error SubPoolFactory__UnknownPool();
+    error SubPoolFactory__PoolHasNotAnnouncedExitYet();
+    error SubPoolFactory__PoolAlreadyAnnouncedExit();
+    error SubPoolFactory__InvalidFastTrackExit();
+    error SubPoolFactory__CannotBillAfterExitDelay();
+    error SubPoolFactory__SolverHasActiveMembership();
+    error SubPoolFactory__SolverNotAMember();
+
+    event UpdateExitDelay(uint256 newDelay);
+    event UpdateBackendUri(address indexed pool, string uri);
+    event SolverPoolDeployed(address indexed solver, address indexed pool);
+    event SolverPoolBilled(address indexed pool, uint256 amt, uint256 cowAmt, uint256 ethAmt, string reason);
+    event AnnounceExit(address indexed pool);
+    event UpdateSolverMembership(address indexed pool, address indexed solver, bool isMember);
+
+    struct SubPoolData {
+        /// The collateral token that pool was initialized with.
+        address collateral;
+        /// The timestamp at which the pool can exit.
+        uint176 exitTimestamp;
+    }
+
+    /// @notice Individual subpool data
+    mapping(address => SubPoolData) public subPoolData;
+    /// @notice Solver backend uri
+    /// @dev    Not stored in the SubPoolData because we don't want to load the string
+    ///         everytime we read the subpooldata in memory.
+    mapping(address => string) public backendUri;
+    /// @notice One to one mappping of solver to a subpool.
+    /// @dev    If a solver already has a subpool of their own, they cannot belong to any other pool.
+    mapping(address => address) public solverBelongsTo;
+    /// @notice the delay between `announceExit` and `exit`.
+    uint256 public exitDelay;
+    address public immutable COW;
+
+    constructor(uint256 exitDelay_, address cow) {
```

Comment:
Nit. It's nicer to have the owner in the constructor so that the deployment of the pool factory can be done by an unsafe "dev" address and the owner can be set to some safe contract (like a multisig with many owners).